src/App.tsx

```diff
@@ -1,14 +1,56 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/DropdownList';
+
+function debounce(callback: (...args: string[]) => void, delay: number) {
+  let timeId = 0;
+
+  return (...args: string[]) => {
+    window.clearTimeout(timeId);
+
+    timeId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
+function filterPeople(people: Person[], appliedQuery: string) {
+  return people.filter((person) => person.name
+    .toLowerCase()
+    .includes(appliedQuery.toLowerCase()));
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
implement this requirement:
![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/c9f70ea3-71e8-42a5-bb70-2c3d9044cd3d)